src/hash/rescue/rpx/mod.rs

```diff
@@ -119,11 +111,11 @@ impl Hasher for Rpx256 {
 
         // if we absorbed some elements but didn't apply a permutation to them (would happen when
         // the number of elements is not a multiple of RATE_WIDTH), apply the RPX permutation. we
-        // don't need to apply any extra padding because the first capacity element contains a
-        // flag indicating whether the input is evenly divisible by the rate.
+        // don't need to apply any extra padding because the second capacity element contains a
+        // flag indicating the number of field elements constituting the last block when the latter
+        // is not divisible by `RATE_WIDTH`.
```

Comment:
I'm not sure this will work. Wouldn't the following sequences of bytes hash to the same value?
```
[1, 2, 3, 4, 5, 6, 7, 8, 0]
[1, 2, 3, 4, 5, 6, 7, 8, 0, 0]
```
When converted to field elements, it would look like so:
```
[Felt::from([1, 2, 3, 4, 5, 6, 7], Felt::from([8, 0])]
[Felt::from([1, 2, 3, 4, 5, 6, 7], Felt::from([8, 0, 0])]
```
I think the padding rule that could work would just append 1 bit to the last field element. Specifically, the above inputs would be transformed as follows:
```
[Felt::from([1, 2, 3, 4, 5, 6, 7], Felt::from([8, 0, 1])]
[Felt::from([1, 2, 3, 4, 5, 6, 7], Felt::from([8, 0, 0, 1])]
```
If the last block is 7 bytes, this shouldn't be a problem as we appending a byte containing value `1` to a 7-byte sequence would still result in a valid field element.

Or are there any issues this scheme?